src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,53 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * 
+ * Represents a Person's address in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
+ *
+ */
+
+public class Contact {
+	
+	public final String contactInfo;
+	private boolean isPrivate;
+	
+	public Contact(String contact, boolean isPrivate, String messageConstraints, String validationRegex) throws IllegalValueException {
+		this.contactInfo = contact.trim();
+		this.isPrivate = isPrivate;
+		
+		if (!checkValidRegex(validationRegex)) {
+			throw new IllegalValueException(messageConstraints);
+		}
+	}
+	
+	
+	
+	private boolean checkValidRegex(String validationRegex) {
+		return this.contactInfo.matches(validationRegex);
+	}
+
+	public boolean isPrivate() {
+		return isPrivate;
+	}
+
+	@Override
+	public String toString() {
+		return contactInfo;
+	}
+
+	@Override
+	public boolean equals(Object other) {
+		return other == this 
+				|| (other instanceof Contact	
+						&& this.contactInfo.equals(((Contact) other).contactInfo)); 
```

Comment:
Note that now an `Email` and `Address` having the same value will be considered equal even if they are of different types.